server/raft.go

```diff
@@ -830,20 +835,21 @@ func (n *raft) ProposeMulti(entries []*Entry) error {
 		return werr
 	}
 	for _, e := range entries {
-		n.prop.push(e)
+		n.prop.push(newProposedEntry(e, _EMPTY_))
 	}
 	return nil
 }
 
 // ForwardProposal will forward the proposal to the leader if known.
 // If we are the leader this is the same as calling propose.
-// FIXME(dlc) - We could have a reply subject and wait for a response
-// for retries, but would need to not block and be in separate Go routine.
 func (n *raft) ForwardProposal(entry []byte) error {
```

Comment:
I can change the function signature, but the struct is still necessary as we need to pass the reply subject through the proposal queue so that we can reply from the `runAsLeader` goroutine once the entries have been stored/sent.

Reply:
These are very rare and feels wasteful to always have an extra 16bytes per proposal..

Reply:
They're going to be a lot more common if we're going to switch to proposal-based deletes, like in #6140, at some point. I am not sure the 16 bytes matters a lot here, things are generally only on the proposal queue for a very short amount of time before they get taken off and written to the log/sent as AEs.

Reply:
That's probably a fair assesment.